extensions/wikia/CloseMyAccount/maintenance/CloseMyAccountMaintenance.php

```diff
@@ -15,80 +18,80 @@ public function __construct() {
-		$res = $dbr->select(
-			'user_properties',
-			[ 'up_user' ],
-			[
-				'up_property' => 'requested-closure',
-				'up_value' => 1,
-				'up_user IN (' . $subQuery . ')',
-			],
-			__METHOD__
-		);
-
-		$usersClosed = [];
+		return $preferenceService->findUsersWithGlobalPreferenceValue( CloseMyAccountHelper::REQUEST_CLOSURE_PREF );
+	}
 
-		$closeReason = 'User requested account closure more than 30 days ago';
+	public function closeAccounts( $users ) {
+		$accountsClosed = 0;
 
-		$wgUser = User::newFromName( 'Wikia' );
+		if ( empty( $users ) ) {
+			return $accountsClosed;
+		}
+
+		$closeAccountHelper = new CloseMyAccountHelper();
 
-		foreach ( $res as $row ) {
-			$userObj = User::newFromId( $row->up_user );
+		foreach ( $users as $userId ) {
+			$userObj = User::newFromId( $userId );
 			$daysRemaining = $closeAccountHelper->getDaysUntilClosure( $userObj );
 
 			if ( $daysRemaining === 0 ) {
-				$this->output( "Closing account {$userObj->getName()}...\n" );
-				$statusMsg1 = '';
-				$statusMsg2 = '';
-				$result = EditAccount::closeAccount( $userObj, $closeReason, $statusMsg1, $statusMsg2, /*$keepEmail = */true );
+				$success = $this->closeUserAccount( $userObj );
+				if ( !$success ) {
+					continue;
+				}
 
-				// Set an option that signifies this user was closed automatically
-				$userObj->setGlobalFlag( 'disabled-by-user-request', true );
-
-				// Cleanup
-				$userObj->setGlobalFlag( 'requested-closure', null );
+				$closeAccountHelper->track( $userObj, 'account-closed' );
 
-				// requested-closure-date is temporarily being stored as both an attribute and a preference.
-				// Make sure to delete from both places. This will be changed to just a preference once the
-				// migration is complete. See SOC-2185
-				$userObj->setGlobalAttribute( 'requested-closure-date', null );
-				$userObj->setGlobalPreference( 'requested-closure-date', null );
+				$accountsClosed++;
+			} else {
+				$this->output( "User {$userObj->getName()} has $daysRemaining days left\n" );
+			}
+		}
 
-				$userObj->saveSettings();
+		return $accountsClosed;
+	}
 
-				$closeAccountHelper->track( $userObj, 'account-closed' );
+	public function closeUserAccount( User $userObj ) {
+		$this->output( "Closing account {$userObj->getName()} ... " );
 
-				$usersClosed[] = $userObj->getName();
-			}
+		$closeReason = 'User requested account closure more than 30 days ago';
+		$statusMsg1 = '';
+		$statusMsg2 = '';
+		$keepEmail = true;
+		$success = EditAccount::closeAccount( $userObj, $closeReason, $statusMsg1, $statusMsg2, $keepEmail );
+
+		if ( !$success ) {
+			$this->output( "failed\n" );
+			return false;
 		}
 
-		$this->output( count( $usersClosed ) . " user accounts closed.\n" );
+		// Set a preference that indicates this user was closed automatically
+		$userObj->setGlobalPreference( CloseMyAccountHelper::DISABLED_BY_USER_PREF, true );
+
+		// Cleanup
+		$userObj->setGlobalPreference( CloseMyAccountHelper::REQUEST_CLOSURE_PREF, null );
```

Comment:
I think there should also be a call following this to nullify the attribute as well (deleted line 80)
```php
$userObj->setGlobalPreference( CloseMyAccountHelper::REQUEST_CLOSURE_PREF, null );
```